pages/scan-results/ScanResults.js

```diff
@@ -192,29 +193,47 @@ class ScanResults extends Component {
     this.setState({ packages })
   }
 
-  handleSortAlphabetic = () => {
+  setParamsAndState = ( sortMode ) => {
     const { packages } = this.state
-    const sortedList = packages.sort((packA, packB) =>
-      packA.name.localeCompare(packB.name),
-    )
+    const query = packages
+        .map(pack => `${pack.name}@${pack.version}`)
+        .join(',')
+    Router.push(`/scan-results?packages=${query}&sortMode=${sortMode}`)
```

Comment:
```suggestion
    Router.replace(`/scan-results?packages=${query}&sortMode=${sortMode}`)
```

We want to replace the history state here so that every sorting change doesn't add a new frame to the history stack.

Reply:
Also, this approach of building query strings manually can prove to be a little brittle.
Consider what happens as time passes by, and we add more query strings to this page – this function would quickly get out of date.

If the intent of this function is just to update the `sortMode` parameter in URLs, consider constraining it using [PLOP](https://en.wikipedia.org/wiki/Principle_of_least_privilege) (Principle of least privilege).

Then this function would keep all existing query params, and only update what it needs. 

```js
  const updatedQuery =  { ...router.query,  sortMode }
  Router.replace(`/scan-results?${queryString.stringify(updatedQuery)}`)
```